autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/hypervisors/xenserver/XenHypervisorTest.java

```diff
@@ -0,0 +1,350 @@
+        setupChangeMasterIfNeeded(true);
+        xenHypervisor.changeMasterIfNeeded(conn, host, "hostUuid");
+        verifyChangeMasterIfNeededCalledMethods(0);
+    }
+
+    @Test
+    public void isLastHostOnPoolTest() throws Exception {
+        Set<Host> setHosts = new HashSet<Host>();
+        setHosts.add(host);
+        setIsLastHostOnPool(setHosts);
+
+        boolean result = xenHypervisor.isLastHostOnPool(conn);
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    public void isLastHostOnPoolTestEmptySet() throws Exception {
+        Set<Host> setHosts = new HashSet<Host>();
+        setIsLastHostOnPool(setHosts);
+
+        boolean result = xenHypervisor.isLastHostOnPool(conn);
+        Assert.assertFalse(result);
+    }
+
+    @Test
+    public void getMasterHostTest() throws Exception {
+        Pool pool = Mockito.mock(Pool.class);
+        pool.getMaster(Mockito.any(Connection.class));
+        Set<Pool> setOfPools = new HashSet<Pool>();
+        setOfPools.add(pool);
+        PowerMockito.mockStatic(Pool.class);
+        PowerMockito.doReturn(setOfPools).when(Pool.class, "getAll", conn);
+
+        xenHypervisor.getMasterHost(conn);
+    }
+
+    @Test(expected = CloudRuntimeException.class)
+    public void getMasterHostTestEmptyPool() throws Exception {
+        Set<Pool> setOfPools = new HashSet<Pool>();
+        PowerMockito.mockStatic(Pool.class);
+        PowerMockito.doReturn(setOfPools).when(Pool.class, "getAll", conn);
+
+        xenHypervisor.getMasterHost(conn);
+    }
+
+    @Test
+    public void changePoolMasterHostTestSameHostAndMasterUuid() throws Exception {
+        setChangePoolMasterHost("hostUuid", true);
+        xenHypervisor.changePoolMasterHost(conn, "hostUuid");
+        verifyChangePoolMasterHostCalledMethods(0, 0);
+    }
+
+    @Test
+    public void changePoolMasterHostTestHostUnreachable() throws Exception {
+        setChangePoolMasterHost("uuid", false);
+        xenHypervisor.changePoolMasterHost(conn, "hostUuid");
+        verifyChangePoolMasterHostCalledMethods(1, 0);
+    }
+
+    @Test
+    public void changePoolMasterHostTestFulExecution() throws Exception {
+        setChangePoolMasterHost("uuid", true);
+        xenHypervisor.changePoolMasterHost(conn, "hostUuid");
+        verifyChangePoolMasterHostCalledMethods(1, 1);
+    }
+
+    @Test
+    public void waitChangePoolMasterHostTest() throws Exception {
+        setupWaitChangePoolMasterHost("uuidDifferentFromHost");
+        xenHypervisor.waitChangePoolMasterHost(host, conn, "uuid");
+    }
+
+    @Test(expected = CloudRuntimeException.class)
+    public void waitChangePoolMasterHostTestExpectCloudRuntimeException() throws Exception {
+        setupWaitChangePoolMasterHost("uuid");
+        xenHypervisor.waitChangePoolMasterHost(host, conn, "uuid");
+        Mockito.verify(threadUtils, Mockito.times(90)).sleepThread(Mockito.anyInt());
+    }
+
+    @Test
+    public void getConnectionTest() throws Exception {
```

Comment:
I think you can annotated this method with "@SuppressWarnings("unchecked")" to suppress a warning.